independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java

```diff
@@ -445,7 +446,7 @@ private CurationResult createAppModelForJar(Path appArtifactPath) {
         final Iterator<AppDependency> depsI = appDeps.iterator();
         while (depsI.hasNext()) {
             final AppArtifact appDep = depsI.next().getArtifact();
-            if (!appDep.getType().equals(AppArtifact.TYPE_JAR)) {
+            if (!appDep.getType().equals(AppArtifactCoords.TYPE_JAR)) {
```

Comment:
What's the reason of this change?

Reply:
https://sonarcloud.io/project/issues?id=gwenneg_quarkus&open=AXAxfQvxgbHBif7sEC6s&resolutions=FIXED&severities=CRITICAL

The `TYPE_JAR` constant is actually defined in `AppArtifactCoords` which is extended by `AppArtifact`.